src/pages/Phase_02.svelte

```diff
@@ -7,36 +7,117 @@ Data needed:
+  import EndowmentScale from "../components/EndowmentScale.svelte";
+  import Rating from "../components/Rating.svelte";
   import Button from "../components/Button.svelte";
 
-  let site = $groupStore.trials[$groupStore.currentTrial].site;
-  let protocol = roundHalf(
-    $groupStore.trials[$groupStore.currentTrial].pain_dur
-  );
+  console.log("userStore", $userStore);
+  console.log("groupStore", $groupStore);
+  console.log("globalVars", globalVars);
 
   const dispatch = createEventDispatcher();
+  let submitted = false;
+  let disableInput = false;
+  let currentQ = 0;
+  let showButton = true;
+  let questions;
 
-  function goToDelivery() {
+  // GET TRIAL DATA
+  // Shared endowment
+  let endowment = $groupStore.trials[$groupStore.currentTrial].endowment;
+  let investedAmount = $groupStore.trials[$groupStore.currentTrial].I_CHOICE;
+  let multiplier = globalVars.multiplier;
+
+  const otherName =
+    $userStore.role === "investor" ? $groupStore.T_name : $groupStore.I_name;
+
+  // Now setup rating scales
+  let i_predict = 0.5 * (investedAmount * multiplier); // investor's rating
+  let t_expect = 0.5 * (investedAmount * multiplier); // trustee's rating
+
+  if ($userStore.role == "investor") {
+    questions = [
+      {
+        questionText: `How much do you think ${$groupStore.T_name} will return to you?`,
+        rating: i_predict,
+        questionType: "other",
+        endowment: investedAmount * multiplier,
+      },
+    ];
+  } else {
+    // For Trustee we ask how much they expect the investor to share with them
+    questions = [
+      {
+        questionText: `How much do you think ${$groupStore.I_name} expects you to return to them?`,
+        rating: t_expect,
+        questionType: "self",
+        endowment: investedAmount * multiplier,
+      },
+    ];
+  }
+
+  console.log("questions", questions);
+
+  async function goto_phase_03() {
+    submitted = true;
+    await saveQData(questions);
     dispatch("to-phase-03");
   }
+
+  async function getNextQuestion() {
+    console.log("currentQ", currentQ);
+    // If they're done answering move to next state
+    if (currentQ === questions.length - 1) {
+      await goto_phase_03();
+    } else {
+      currentQ = currentQ + 1;
+      showButton = true;
+      switchToRatingScale = false;
+      disableInput = false;
+    }
+  }
 </script>
 
-<div class="flex h-screen text-center">
-  <div class="m-auto">
-    {#if $userStore.role === "receiver"}
-      <h1 class="mb-4 text-2xl">
-        Experimenter please place the thermode on site {site} and use protocol {protocol}
-      </h1>
-      <Button on:click={goToDelivery}>Ready</Button>
-    {:else}
-      <h1 class="mb-4 text-2xl">Waiting for experimenter to place thermode</h1>
-    {/if}
+{#if submitted}
+  <Loading text={"Waiting for your partner..."} />
+{:else}
+  <div class="w-3/5 mx-auto">
+    <div class="min-w-full pb-32 text-center">
+      <div class="my-10">
```

Comment:
This is where I imagine a prop for the EndowmentScale would be helpful. Because I'm not sure they need to see the "shared endowment" and "multiplier" info when they're making their expectation ratings.

Reply:
thanks! and agreed. i actually want to add a screen before each phase to tell them 'you're about to enter x trial, endowment: y, multiplier: z' so will def move that stuff around

Reply:
https://github.com/cosanlab/trust-game/commit/da9dc2d45acff5885e6ab32c2ecb2534360e86ab